### DISCUSSION TITLE
Hacker News Article

### DISCUSSION THREAD

**Discussion by grantcarthew** [grantcarthew/awesome-unique-id#13] [General]
Hey @tzwel, I thought you might be interested in this?

https://news.ycombinator.com/item?id=40582365

**Comment by tzwel** [comment]
Yup! I read this one some time ago!
Here are some interesting reads about random numbers I found lately too:
- https://research.kudelskisecurity.com/2020/07/28/the-definitive-guide-to-modulo-bias-and-how-to-avoid-it
- https://www.pcg-random.org/posts/bounded-rands.html (this one is golden)

I'm looking for more stuff like this, numbers are fascinating. Have you got more stuff like this?

Edit:
Actually, I needed a simple ID generator for my own purposes (user account IDs in my system) and that Hackernews post gave me an idea on approaching this. I open sourced it as [breezeid](https://github.com/tzwel/BreezeID) (still WIP).  Way simpler than uuid which adds unnecessary complexity in my opinion

**Comment by grantcarthew** [comment]
Breeze ID looks great. I had a laugh as the profanities array.
I'm not really developing in JS much at this time. Mostly C++ with the Qt Framework. I'm also leaning towards using [PocketBase](https://pocketbase.io/) which means IDs are not in my field of view. No extra resources sorry.

Back when I made [Scalable Blob Store](https://github.com/grantcarthew/node-scalable-blob-store) I used Eric Elliott's CUID package which is now [Cuid2](https://github.com/paralleldrive/cuid2). I've not compared it to newer id generators though.

**Reply by tzwel** [reply]
Have you seen Cuid2's code? I find it interesting that it uses just `Math.random` and `Date.now()` for its entropy (and a counter). Seems insecure but nobody had a problem with it?

**Reply by grantcarthew** [reply]
I have a huge amount of respect for Eric Elliott and at the time, his research was extensive. I've not analyzed it recently.
